src/views/Player.vue

```diff
@@ -105,15 +115,32 @@ export default defineComponent({
       hasQuizEnded: false, // whether the quiz has ended - only valid for quizType = assessment
       sessionId: "", // id of the session created for a user-quiz combination
     });
+
+    const requiresAuth = computed(() => {
+      return props.userId != null && props.apiKey != null;
+    });
+
+    async function getAuth() {
+      if (requiresAuth.value) {
+        UserAPIService.checkAuthToken(props.apiKey).catch(() => {
+          router.replace({
+            name: "403",
+          });
+        });
+      }
+    }
+
+    getAuth();
```

Comment:
you don't need to define a new function - you can directly write:
```suggestion
    UserAPIService.checkAuthToken(props.apiKey).catch(() => {
          router.replace({
            name: "403",
          });
        }); 
```